File: src/main/java/com/letgo/book/infrastructure/persistence/mapping/BookIdType.java
package com.letgo.book.infrastructure.persistence.mapping;

import com.letgo.book.domain.BookId;
import org.hibernate.type.AbstractSingleColumnStandardBasicType;
import org.hibernate.type.descriptor.sql.CharTypeDescriptor;

final public class BookIdType extends AbstractSingleColumnStandardBasicType<BookId> {
    public BookIdType() {
        super(new CharTypeDescriptor(), new BookIdDescriptor());
    }

    @Override
    public String getName() {
        return "BookId";
    }
}

File: src/main/kotlin/com/letgo/book/infrastructure/persistence/mapping/BookIdType.kt
package com.letgo.book.infrastructure.persistence.mapping

import com.letgo.book.domain.BookId
import org.hibernate.type.AbstractSingleColumnStandardBasicType
import org.hibernate.type.descriptor.sql.CharTypeDescriptor

class BookIdType : AbstractSingleColumnStandardBasicType<BookId>(CharTypeDescriptor(), BookIdDescriptor()) {
    override fun getName(): String = "BookId"
}
